source/agora/common/BlockStorage.d

```diff
@@ -761,6 +788,80 @@ public class BlockStorage : IBlockStorage
         {
         }
     }
+
+    /*******************************************************************************
+
+        Read the bytes to calculate the checksum
+
+        Params:
+            data = file data bytes
+
+        Returns:
+            calculate the checksum
+
+    *******************************************************************************/
+
+    private ubyte[] checksum (const ubyte[] data) @safe nothrow
+    {
+        auto crc32 = new CRC32Digest();
+        crc32.put(data);
+        return crc32.finish();
+    }
+
+    /*******************************************************************************
+
+        Read the MemFile bytes and calculate them as checksums and
+        compare them with the expected checksum.
+
+        Returns:
+            Returns true if correct, otherwise returns false
+
+    *******************************************************************************/
+
+    private void validateChecksum () @trusted nothrow
+    {
+        try
+        {
+            const ubyte [] expected = cast(ubyte[])this.file[0 .. ChecksumSize];
+            const ubyte [] fileData = cast(ubyte[])this.file[ChecksumSize .. MapSize];
+            const actual = this.checksum(fileData);
+            if (expected != actual)
+            {
+                stderr.writefln("[ERROR] %s Block file is corrupted and will be terminated." , this.getFileName(this.file_index));
+                abort();
+            }
+        }
+        catch (Exception ex)
+        {
+            this.writeLog("BlockStorage.compareChecksum: ", ex);
```

Comment:
If the checksum is read, but it doesn't match, we abort. But if the checksum fails to be read, we just ignore the error. We should be consistent and abort in all cases.

Reply:
add abort();

Reply:
Use `assert(0)` instead of `abort()`

Reply:
Applied the suggestion